### ISSUE TITLE
Wrong type definitions when using with Ember

### ISSUE THREAD

**Issue by josemarluedke** [glimmerjs/glimmer.js#179]
Using `@glimmer/component` version 0.14.0-alpha.3 with Ember (canary) brings the wrong type definitions for TypeScript users.

## Current behavior

The types that get used are referent to the glimmer.js implementation of the component.

That implementation is not complaint with  Glimmer Components [RFC](https://github.com/emberjs/rfcs/pull/416). It includes additional hooks and properties such as  `didInsertElement, bounds, element, didUpdate` on the component class.

This gives wrong type errors when writing components in TS as well wrong auto complete suggestions.

### Reproduction

1) Clone [this app](https://github.com/josemarluedke/--ts-app-glimmer)
2) Open a tool that has TS lang server such as VS code and play around

## Expected

Types that reflect actual implementation of Glimmer Components RFC.

**Comment by chriskrycho** [comment]
I just checked and the correct info is on `master`; we just need another alpha (or beta?) release cut.

**Comment by josemarluedke** [comment]
I want to clarify that the this issue is referring to the type definitions of the component class.

See following images:

![Screen Shot 2019-03-12 at 1 38 48 PM](https://user-images.githubusercontent.com/230476/54289864-2c60cc00-4567-11e9-8e55-a60b754bc8e7.png)
![Screen Shot 2019-03-12 at 1 39 06 PM](https://user-images.githubusercontent.com/230476/54289866-2c60cc00-4567-11e9-9db8-71fe7eb899e6.png)

Per Glimmer RFC, we should not have `didInsertElement` or `element`. The VS Code auto complete suggests the wrong thing because it is pulling the wrong type for the component.

It uses [this file](https://github.com/glimmerjs/glimmer.js/blob/master/packages/%40glimmer/component/src/component.ts) instead of [this file](https://github.com/glimmerjs/glimmer.js/blob/master/packages/%40glimmer/component/addon/-private/component.ts).

**Comment by rwjblue** [comment]
Chatted about this a bit in discord, for now the easiest fix is to override in `tsconfig.json`.

@chriskrycho / @josemarluedke - Could one of y'all paste the snippet?

**Comment by chriskrycho** [comment]
In your `tsconfig.json`'s `compilerOptions` hash:

```json
    "paths": {
      "@glimmer/component": [
        "node_modules/@glimmer/component/addon"
      ]
    },
```

**Comment by jamescdavis** [comment]
This should be fixed in `@glimmer/component` 2.0

See : https://github.com/glimmerjs/glimmer.js/pull/235

**Comment by locks** [comment]
PR mentioned above has been merged.
